packages/react-form-state/src/FormState.ts

```diff
@@ -122,19 +122,29 @@ export default class FormState<
     };
   }
 
+  public validateForm() {
+    return new Promise(resolve => {
+      this.setState(runAllValidators, () => resolve());
+    });
+  }
+
   private get dirty() {
     return this.state.dirtyFields.length > 0;
   }
 
   private get valid() {
-    const {errors, fields} = this.state;
+    const {errors} = this.state;
 
-    const fieldsWithErrors = Object.keys(fields).filter(fieldPath => {
-      const {error} = fields[fieldPath];
-      return error != null;
-    });
+    return !this.hasClientErrors && errors.length === 0;
+  }
+
+  private get hasClientErrors() {
```

Comment:
should this be public as well?

Reply:
I can't think of a case where I'd need this instead of `valid`.